src/core/tests/type_prop/binary_elementwise.cpp

```diff
@@ -445,3 +445,316 @@ TEST(type_prop, logic_arith_compare_partial_et) {
+        }
+    }
+}
+
+TEST(type_prop, interval_value_propagation_mul) {
+    PartialShape op_shape{Dimension(-1), Dimension(4, -1), Dimension(-1, 6), Dimension(5, 7), Dimension(9, 10), 15};
+    {  // const rhs
+        const auto const_op = op::Constant::create(element::f32, {6}, {7, 6, 5, 4, 3, 2});
+        const auto reshape = createReshapeSubgraph<op::v1::Multiply>(op_shape, const_op);
+        EXPECT_EQ(reshape->get_element_type(), element::f32);
+        EXPECT_EQ(reshape->get_output_partial_shape(0),
+                  PartialShape({-1, -1, Dimension(-1, 30), Dimension(20, 28), Dimension(27, 30), 30}));
+    }
+    {  // const lhs
+        const auto const_op = op::Constant::create(element::f32, {6}, {7, 6, 5, 4, 3, 2});
+        const auto reshape = createReshapeSubgraph<op::v1::Multiply>(op_shape, const_op, false);
+        EXPECT_EQ(reshape->get_element_type(), element::f32);
+        EXPECT_EQ(reshape->get_output_partial_shape(0),
+                  PartialShape({-1, -1, Dimension(-1, 30), Dimension(20, 28), Dimension(27, 30), 30}));
+    }
+    {  // const rhs - result lower than 0
+        PartialShape op_shape{Dimension(5, 7)};
+        const auto const_op = op::Constant::create(element::f32, {1}, {-3});
+        try {
+            const auto reshape = createReshapeSubgraph<op::v1::Multiply>(op_shape, const_op);
+            FAIL() << "Incorrect dimenion was not detected.";
+        } catch (const NodeValidationFailure& error) {
+            EXPECT_HAS_SUBSTRING(error.what(), std::string("Dim size cannot be less than -1"));
+        } catch (...) {
+            FAIL() << "Incorrect dimenion was not detected, different error type was thrown.";
+        }
+    }
+    {  // const lhs - result lower than 0
+        PartialShape op_shape{Dimension(5, 7)};
+        const auto const_op = op::Constant::create(element::f32, {1}, {-3});
+        try {
+            const auto reshape = createReshapeSubgraph<op::v1::Multiply>(op_shape, const_op, false);
+            FAIL() << "Incorrect dimenion was not detected.";
+        } catch (const NodeValidationFailure& error) {
+            EXPECT_HAS_SUBSTRING(error.what(), std::string("Dim size cannot be less than -1"));
+        } catch (...) {
+            FAIL() << "Incorrect dimenion was not detected, different error type was thrown.";
+        }
+    }
+}
+
+TEST(type_prop, interval_value_propagation_div) {
+    {  // const rhs
+        PartialShape op_shape{Dimension(8, 16), Dimension(9, 30), 15};
+        const auto const_op = op::Constant::create(element::f32, {3}, {4, 3, 5});
+        const auto reshape = createReshapeSubgraph<op::v1::Divide>(op_shape, const_op);
+        EXPECT_EQ(reshape->get_element_type(), element::f32);
+        EXPECT_EQ(reshape->get_output_partial_shape(0), PartialShape::dynamic(3));
+
+        // TODO: Why not:
+        // EXPECT_EQ(reshape->get_output_partial_shape(0), PartialShape({Dimension(2, 8), Dimension(3, 10), 3}));
```

Comment:
Such case for Divide, leads to fully dynamic dimensions only.
@jane-intel Is it correct?

Reply:
It is not correct. The expected result from the comment should be returned

Reply:
Current solution is to change `get_lower/upper_value` to `evaluate_lower/upper_bound` in the `Divide` custom `evaluate_bound` function (the same approach as in `interval_bound_evaluator`, common for other `BinaryElementwiseArithmetic` ops).

The change is visible with the commit: https://github.com/openvinotoolkit/openvino/pull/13390/commits/e771dbd74e071ca80923ba8a26525dd1699ea49e